UPGRADING.md

```diff
@@ -0,0 +1,117 @@
+
+There will no doubt be breakages in using calls to now non-existent endpoints (as populated by the metadata) if you are upgrading your nodes to Substrate 2.x. Substrate 2.x has had a number of internal changes, where new modules and features are introduced (such as `babe` and `technicalCommittee`), some modules have been renamed (such as `contract` -> `contracts`) and modules such as `session` has been reworked to a large degree.
+
+To cater for both 1.x and 2.x support, the [@polkadot/api-derive](packages/api-derive) endpoints, do feature detection for the node type and should continue working as-is. Additionally a number of new derives have been added, specifically around elections.
+
+### Type renames
+
+To better align with the actual types from the metadata, and avoid (too much) context switching, some types from the `@polkadot/types` hasve been renamed. These include -
+
+- `Vector` -> `Vec`
+- `U{8|16|32|64|128|256}` have been removed, only the lowercase version of these remain, i.e. `u32`.
+
+### Type usage
+
+The [@polkadot/api][packages/api] has always handled the conversion of types for parameters when making calls or queries. For example, when making a transfer to `BOB` (address), any of the following is valid -
+
+- `api.tx.balances.transfer(BOB, 12345)` - value specied as a number
+- `api.tx.balances.transfer(BOB, '12345')` - value specied as a string
+- `api.tx.balances.transfer(BOB, '0x3039')` - value specied as a hex
+- `api.tx.balances.transfer(BOB, new BN(12345))` - value specied as a [BN](https://github.com/indutny/bn.js/)
+
+Internally the API will take the input and convert the value into a `Balance`, serialize it using the SCALE codec and transfer it to the node. In some cases users would construct the `Balance` type manually, by importing the class and calling `new` on it. This last approach has now been removed, and where classes are still available (limited reach), discouraged.
+
+First the rationale behind this - in all cases Substrate is very flexible, so while Polkadot (and the Substrate base), define `type Balance = u128`, this can be different between chains. (This also applies to the majority of built-in supported types). As such, type construction should be done via the actual registered types.
+
+```js
+// this is applicable everywhere, import the type creator, using the registry
+import { createType } from '@polkadot/types';
+
+// construct the Balance, of type Balance (type is inferred and available with TS)
+const value = createType('Balance', 12345);
+
+// use value here as you normally would
+...
+```
+
+The impact of this will be noticable, if you have been importing the old-style type classes from `@polkadot/types`, those imports are not available anymore. For creation, just pass everything through the `createType`.
```

Comment:
```suggestion
The impact of this will be noticeable, if you have been importing the old-style type classes from `@polkadot/types`, those imports are not available anymore. For creation, just pass everything through the `createType`.
```